### PR TITLE
feat(pr-comment): update notifiers.go interface

### PR DESCRIPTION
**Pull request by cristhian-castaneda**
Update the notifiers interface so we can extend the usability and send more content to their implementations.

e.g. if some notifier needs some other information apart of the org, repo, path, notifications now we can send it via the `content` map.

### CODE REVIEW

File: pkg/dinghyfile/builder.go [hunk 1]
```diff
@@ -538,3 +538,16 @@ func getParams(regEx, url string) (paramsMap map[string]string) {
 	}
 	return paramsMap
 }
+
+func (b *PipelineBuilder) getContent() map[string]interface{} {
+	content := map[string]interface{}{}
+
+	logEvent, err := b.Logger.GetBytesBuffByLoggerKey(log.LogEventKey)
+	if err != nil {
+		b.Logger.Info(fmt.Sprintf("there was an error trying to get content: %v", err))
```
Comment (Jossuecito) [inline pkg/dinghyfile/builder.go]: I think this needs to change to error or warning

File: pkg/dinghyfile/builder.go [hunk 2]
```diff
@@ -538,3 +538,16 @@ func getParams(regEx, url string) (paramsMap map[string]string) {
 	}
 	return paramsMap
 }
+
+func (b *PipelineBuilder) getContent() map[string]interface{} {
+	content := map[string]interface{}{}
```
Comment (Jossuecito) [inline pkg/dinghyfile/builder.go:543]: double {}

**Review by dogonthehorizon — APPROVED** [review]

File: pkg/dinghyfile/builder.go (hunk 2 above)
Comment (dogonthehorizon) [reply pkg/dinghyfile/builder.go:543]: I think it's fine actually, this reads like "instantiate a map of string to interface"

File: pkg/dinghyfile/builder.go [hunk 3]
```diff
@@ -538,3 +538,16 @@ func getParams(regEx, url string) (paramsMap map[string]string) {
...
+
+	logEvent, err := b.Logger.GetBytesBuffByLoggerKey(log.LogEventKey)
+	if err != nil {
+		b.Logger.Info(fmt.Sprintf("there was an error trying to get content: %v", err))
+	} else {
+		content["logevent"] = logEvent.String()
+	}
+	content["raw"] = b.PushRaw
```
Comment (Jossuecito) [inline pkg/dinghyfile/builder.go]: Can we change this to rawdata

File: pkg/dinghyfile/builder_test.go [hunk 4]
```diff
@@ -1253,4 +1249,100 @@ pipelines:
...
+			want: map[string]interface{}{
+				"raw": map[string]interface{}{
+					"head_commit": map[string]interface{}{
+						"id": "a5fc63bd5a8bdb342d1e83933a5b5c99010e61e4",
+					},
+				},
+			},
+		},
+		{
+			name: "Content should return a map populated with 'raw' and 'logevent' properties",
```
Comment (Jossuecito) [inline pkg/dinghyfile/builder_test.go]: Only github uses PushRaw AFAIK, so maybe we should add one extra test also when it's nil

File: pkg/dinghyfile/builder_test.go [hunk 5]
```diff
@@ -1253,4 +1249,100 @@ pipelines:
...
+			fields: fields{
+				Logger: func() log.DinghyLog {
+					memLog := &bytes.Buffer{}
+					memLog.Write([]byte("test"))
+					l := log.DinghyLogs{Logs: map[string]log.DinghyLogStruct{
```
Comment (Jossuecito) [inline pkg/dinghyfile/builder_test.go]: You can call NewDinghylog()  from test_commmon

File: pkg/dinghyfile/builder.go [hunk 6]
```diff
@@ -538,3 +538,16 @@ func getParams(regEx, url string) (paramsMap map[string]string) {
 	}
 	return paramsMap
 }
+
+func (b *PipelineBuilder) getContent() map[string]interface{} {
```
Comment (Jossuecito) [inline pkg/dinghyfile/builder.go]: Maybe getNotificationContent would be easier to identify

**Review by Jossuecito — APPROVED** [review]

File: pkg/dinghyfile/builder_test.go (hunk 5 above)
Comment (cristhian-castaneda) [reply pkg/dinghyfile/builder_test.go]: Created another mock function to instantiate a Dinghy Log with predefined content, we could need it afterward.
